slither/detectors/erc/erc20/arbitrary_send_erc20.py

```diff
@@ -27,69 +59,57 @@ def no_permit_results(self) -> List[Node]:
         """Finds instances of (safe)transferFrom that do not use msg.sender or address(this) as from parameter."""
-        for node in nodes:
-            for ir in node.irs:
-                if (
-                    isinstance(ir, HighLevelCall)
-                    and isinstance(ir.function, Function)
-                    and ir.function.solidity_signature == "transferFrom(address,address,uint256)"
-                    and not (
-                        is_dependent(
-                            ir.arguments[0],
-                            SolidityVariableComposed("msg.sender"),
-                            node.function.contract,
-                        )
-                        or is_dependent(
-                            ir.arguments[0],
-                            SolidityVariable("this"),
-                            node.function.contract,
-                        )
-                    )
-                ):
-                    results.append(ir.node)
-                elif (
-                    isinstance(ir, LibraryCall)
-                    and ir.function.solidity_signature
-                    == "safeTransferFrom(address,address,address,uint256)"
-                    and not (
-                        is_dependent(
-                            ir.arguments[1],
-                            SolidityVariableComposed("msg.sender"),
-                            node.function.contract,
-                        )
-                        or is_dependent(
-                            ir.arguments[1],
-                            SolidityVariable("this"),
-                            node.function.contract,
-                        )
-                    )
+
+        if (
+            isinstance(ir, HighLevelCall)
+            and isinstance(ir.function, Function)
+            and ir.function.solidity_signature == "transferFrom(address,address,uint256)"
+            and ir.arguments[0]
+            not in (SolidityVariableComposed("msg.sender"), SolidityVariable("this"))
+        ):
+            results.append(ir.node)
+        elif (
+            isinstance(ir, LibraryCall)
+            and ir.function.solidity_signature
+            == "safeTransferFrom(address,address,address,uint256)"
+            and ir.arguments[1]
+            not in (SolidityVariableComposed("msg.sender"), SolidityVariable("this"))
+        ):
+            results.append(ir.node)
+
+    def _detect_arbitrary_from(self, instructions: List[Operation]):
+        has_permit = False
+        for ir in instructions:
+            if isinstance(ir, InternalCall):
+                self.simplify_and_analyze(ir.function.nodes)
+            # High level calls (view functions) can also be `StateVariable`
+            if isinstance(ir, HighLevelCall) and isinstance(ir.function, Function):
+                sig = ir.function.solidity_signature
+                has_permit |= sig == "permit(address,address,uint256,uint256,uint8,bytes32,bytes32)"
```

Comment:
There is no guarantee on the order of the IR (the list is created by iterating over the nodes - which don't have an order, as they coming from a graph).  
So I think `has_permit` should be assigned ahead of time, or explored through a recursive path exploration.

For example
```solidity
if(..){
   token.permit
   return;
}
token.transferFrom()
```

Here `has_permit` might be set to true, or to false when `token.transferFrom` is reached; it depends on the order of the nodes in `func.nodes`